components/ai_chat/core/browser/associated_content_manager.cc

```diff
@@ -131,7 +130,17 @@ void AssociatedContentManager::AddContent(AssociatedContentDelegate* delegate,
     // fetch because its a bit of an edge case, and there are no real
     // consequences of not having the content (except for the content not being
     // attached).
-    delegate->GetContent(base::DoNothing());
+    delegate->GetContent(base::BindOnce(
+        [](base::WeakPtr<AssociatedContentManager> self, PageContent content) {
+          if (!self) {
+            return;
+          }
+
+          // Note: |is_video| may have changed so we need to notify the
+          // conversation.
+          self->conversation_->OnAssociatedContentUpdated();
```

Comment:
This change should be reflected in `//brave/components/ai_chat/core/browser/associated_content_manager_unittest.cc` that when `AddContent` is called and verify that observer OnAssociatedContentUpdated is notified.